distributed/utils_test.py

```diff
@@ -1768,7 +1768,7 @@ def check_thread_leak():
 
             bad_thread = bad_threads[0]
             call_stacks = profile.call_stack(sys._current_frames()[bad_thread.ident])
-            assert False, (bad_thread, call_stacks)
+            assert False, (bad_thread, call_stacks)  # noqa: B011
```

Comment:
Why ignore here instead of raising an error?

Reply:
Honestly, I don't mind raising an error either. My line of thinking was as follows: From my understanding, the rule is targeted toward production code that should always throw, not tests. Therefore, in #6809, I propose to ignore **B011** for all tests since `assert` statements are perfectly fine in there. (I'm happy to iterate on that.) As `check_thread_leak` is a helper function for tests, it felt like the same rules applied to it as to all other tests for which we ignore **B011**.

Reply:
Thanks for the additional context. I don't have strong feelings either way. Let's go with what you have here -- it's easy enough to follow-up and remove the ignore if other have different thoughts on the topic